pytorch_lightning/strategies/strategy.py

```diff
@@ -51,7 +51,8 @@ def __init__(
         self._model: Optional[Module] = None
         self.checkpoint_io = checkpoint_io
         self.precision_plugin = precision_plugin
-        self.optimizers: List[Optimizer] = []
+        self._optimizers: List[Optimizer] = []
+        self._cached_lightning_optimizers: Dict[int, LightningOptimizer] = {}
```

Comment:
We could also not cache this at all and re-generate it every time it's necessary (num training batches * num optimizers).

It would simplify the code, but not sure if it's a good idea.